src/neuroconv/tools/nwb_helpers/_dataset_configuration.py

```diff
@@ -0,0 +1,195 @@
+"""Collection of helper functions related to configuration of datasets dependent on backend."""
+from typing import Iterable, Literal, Union
+
+import h5py
+import numpy as np
+import zarr
+from hdmf import Container
+from hdmf.data_utils import DataChunkIterator, DataIO, GenericDataChunkIterator
+from hdmf.utils import get_data_shape
+from hdmf_zarr import NWBZarrIO
+from pynwb import NWBHDF5IO, NWBFile, TimeSeries
+from pynwb.base import DynamicTable
+
+from ._models._base_models import DatasetConfiguration, DatasetInfo
+from ._models._hdf5_models import HDF5BackendConfiguration, HDF5DatasetConfiguration
+from ._models._zarr_models import ZarrBackendConfiguration, ZarrDatasetConfiguration
+from ..hdmf import SliceableDataChunkIterator
+
+BACKEND_TO_DATASET_CONFIGURATION = dict(hdf5=HDF5DatasetConfiguration, zarr=ZarrDatasetConfiguration)
+BACKEND_TO_CONFIGURATION = dict(hdf5=HDF5BackendConfiguration, zarr=ZarrBackendConfiguration)
+
+
+def _get_mode(io: Union[NWBHDF5IO, NWBZarrIO]) -> str:
+    """NWBHDF5IO and NWBZarrIO have different ways of storing the mode they used on a path."""
+    if isinstance(io, NWBHDF5IO):
+        return io.mode
+    elif isinstance(io, NWBZarrIO):
+        return io._ZarrIO__mode
+
+
+def _is_value_already_written_to_file(
+    candidate_dataset: Union[h5py.Dataset, zarr.Array],
+    backend: Literal["hdf5", "zarr"],
+    existing_file: Union[h5py.File, zarr.Group, None],
+) -> bool:
+    """
+    Determine if the neurodata object is already written to the file on disk.
+
+    This object should then be skipped by the `get_io_datasets` function when working in append mode.
+    """
+    return (
+        isinstance(candidate_dataset, h5py.Dataset)  # If the source data is an HDF5 Dataset
+        and backend == "hdf5"  # If working in append mode
+        and candidate_dataset.file == existing_file  # If the source HDF5 Dataset is the appending NWBFile
+    ) or (
+        isinstance(candidate_dataset, zarr.Array)  # If the source data is an Zarr Array
+        and backend == "zarr"  # If working in append mode
+        and candidate_dataset.store == existing_file  # If the source Zarr 'file' is the appending NWBFile
+    )
+
+
+def _parse_location_in_memory_nwbfile(current_location: str, neurodata_object: Container) -> str:
```

Comment:
'location' is the field name as referenced by the NWB Inspector data models, so I'd rather be consistent with that

'path' in NeuroConv is also always used in reference to file system entities so I'd rather use a distinct word to avoid confusion with 'paths within a file' as much as possible though I know it's possible to refer to these values in that way (even in HDF5)

I'd probably go with 'find', since I interpret 'get' to mean a fetcher of a 'set', and 'build' would mean the same to me here as 'parse'

Was there a reason why 'parse' didn't communicate it clearly enough?

Reply:
Oh dang, I forgot about that

How about `location_in_file`?

Reply:
Yep. That makes sense to me.

Reply:
OK renamed